src/Components/CommonComponent/Table/Table.tsx

```diff
@@ -22,133 +23,229 @@ export interface StudentProfile {
+                            <img
+                              src="/assests/Tabledelete.svg"
+                              alt="studentProfileDelete"
+                              className={styles?.cell_icon}
+                            ></img>
+                          </IconButton>
+                          <IconButton onClick={() => handleEdit(studentData)}>
+                            <img
+                              src="/assests/eye.png"
+                              alt="studentProfile"
+                              className={`${styles?.cell_icon} ${styles?.eye_icon}`}
+                            ></img>
+                          </IconButton>  
+                        </>
+                      }
+                    </TableCell>
+                  </TableRow>
+                );
+              }else if(type === "SPECTATOR"){
+                console.log("SPECTATOR => STUDENTDATA AND INDEX AND SORTEDDATA AND TYPE 2.1==>",studentData,index,sortedData,type)
+
+                return (
+                  <TableRow className={styles?.table_rowdata} key={index}>
+                    <TableCell className={styles?.table_cell}>{studentData?.fullName}</TableCell>
+                    <TableCell className={styles?.table_cell}>{studentData?.userName || "--"}</TableCell>
+                    <TableCell className={styles?.table_cell}>{studentData?.email}</TableCell>
+                    <TableCell className={styles?.table_cell}>
+                      {
+                        <>
+                          <IconButton onClick={() => handleEdit(studentData)}>
+                            <img
+                              src="/assests/editIcon.svg"
+                              alt="studentProfileEdit"
+                              className={styles?.cell_icon}
+                            ></img>
+                          </IconButton>
+                          <IconButton onClick={() => {
+                            console.log("onClickHandler studentData?._id",studentData?._id)
+                            deleteroom(studentData?.userUuid
+                              )}}>
+                            <img
+                              src="/assests/Tabledelete.svg"
+                              alt="studentProfileDelete"
+                              className={styles?.cell_icon}
+                            ></img>
+                          </IconButton>
+                        </>
+                      }
+                    </TableCell>
+                  </TableRow>
+                );
+              }
+              else if(type === "USERS"){
+                console.log("USERS => STUDENTDATA AND INDEX AND SORTEDDATA AND TYPE 2.1==>",studentData,index,sortedData,type)
+
+                return (
+                  <TableRow className={styles?.table_rowdata} key={index}>
+                    <TableCell className={styles?.table_cell}>{studentData?.fullName}</TableCell>
```

Comment:
is it possible to reuse a single table component for users and spectators?

Reply:
resolved